test/common/common/key_value_store_test.cc

```diff
@@ -0,0 +1,77 @@
+#include <memory>
+#include <string>
+
+#include "source/common/common/key_value_store_base.h"
+
+#include "test/mocks/event/mocks.h"
+#include "test/test_common/environment.h"
+#include "test/test_common/file_system_for_test.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::NiceMock;
+
+namespace Envoy {
+namespace {
+
+class KeyValueStoreTest : public testing::Test {
+protected:
+  KeyValueStoreTest() : filename_(TestEnvironment::temporaryPath("key_value_store")) {
+    TestEnvironment::removePath(filename_);
+    createStore();
+  }
+
+  void createStore() {
+    store_.reset();
```

Comment:
nit: This line is not needed because the assignment on the next line will take care of destroying any existing object (just like reset())

Reply:
fair point